phtree/common/filter.h

```diff
@@ -150,6 +150,132 @@ class FilterAABB {
     const CONVERTER converter_;
 };
 
+/*
+ * The paraboloid filter can be used to query a point tree for an axis-aligned paraboloid.
+ * This is useful for, e.g., doing a range check for a projectile with a fixed launch speed.
+ * Also see https://en.wikipedia.org/wiki/paraboloid_of_safety for this application.
+ */
+template <
+    typename CONVERTER = ConverterIEEE<3>,
+    typename DISTANCE = DistanceEuclidean<3>>
+class FilterParaboloid {
+    using KeyExternal = typename CONVERTER::KeyExternal;
+    using KeyInternal = typename CONVERTER::KeyInternal;
+    using ScalarInternal = typename CONVERTER::ScalarInternal;
+    using ScalarExternal = typename CONVERTER::ScalarExternal;
+
+  public:
+    /*
+     * @param axis The axis along which the paraboloid is aligned.
+     * @param vertex The vertex of the paraboloid.
+     * @param scale_factor Scales the paraboloid along the axis. A positive scale factor
```

Comment:
Can you maybe give an example or explain what the scale factor does? The sign obviously affects the direction. But why is it a 'double'? Does it have any other effect or could it be a 'bool'?

Reply:
It's the scale factor in the quadratic term. [This image](https://d138zd1ktt9iqe.cloudfront.net/media/seo_landing_files/mohit-uniyal-vertical-scaling-04-1609254123.png) shows two parabolas with scale factors 1 and 2. A negative value flips the parabola.

Reply:
So basically for vertex=origin we get `y=scale_factor * x^2`? Maybe you could add a sentence to the doc?